[ML] Fix custom index name settings, functional tests for APM Latency Correlation.

walterra opened the pull request:
## Summary

Follow up to #99905.

- [x] Fixes #105401.
- [x] Requires the fixes in #104644 to be merged to pass.

Adds first functional tests for APM Latency Correlations code.

- **Fix:** The log log chart's Y axis would only start at `1` by default which hides results for small datasets like the ones used in these tests. Starting the axis at `0` isn't supported to log based ones so we're setting it to just a small value above 0.
- **Fix:** Instead of the hard coded `apm-*` index we passed on from the client, we now correctly consider APM's custom settings.

Discuss/Questions:

- I copied the overall test setup from the spaces tests where the full UI is available. Not sure if there would be a simpler way maybe using a superuser? Do you think it's ok like that?
- Debug Log Helper: If you enable `observability:enableInspectEsQueries` in advanced settings, this will output the async service's log we pass on to the client in the flyout.
- Is `'apm_oss.transactionIndices'` the correct index to query, do we need to consider others?
- I added code in `plugin.ts` to pass on `includeFrozen`, we're not making use of it yet though, is this something we want to fix for 7.14 too?

<img src="https://user-images.githubusercontent.com/230104/125277607-ecb27680-e311-11eb-9317-1dbc7ce850db.png" width="50%" />

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

walterra committed:
apm correlations functional tests

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/)
* Commit: 90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/execution/node/428/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/","number":"137794","commit":"90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5"}]}
PIPELINE-->

walterra committed:
fix types.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/)
* Commit: b8b3c9652af3fa0f0313ffe97ad40e523c917451
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/b8b3c9652af3fa0f0313ffe97ad40e523c917451)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/execution/node/716/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/execution/node/825/log)

### History
* :broken_heart: [Build #137794](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/) failed 90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/","number":"137794","commit":"90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/","number":"137804","commit":"b8b3c9652af3fa0f0313ffe97ad40e523c917451"}]}
PIPELINE-->

walterra committed:
fix passing on custom apm index names.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/)
* Commit: 83b6084f86b52defc9eab0ee543a8076e276febd
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/execution/node/428/log)

### History
* :broken_heart: [Build #137804](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/) failed b8b3c9652af3fa0f0313ffe97ad40e523c917451
* :broken_heart: [Build #137794](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/) failed 90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/","number":"137794","commit":"90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/","number":"137804","commit":"b8b3c9652af3fa0f0313ffe97ad40e523c917451"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/","number":"138162","commit":"83b6084f86b52defc9eab0ee543a8076e276febd"}]}
PIPELINE-->

walterra committed:
fix types

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/)
* Commit: 733a971301d3a42cc8e7cb7ec9ee8536597cfafb
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/733a971301d3a42cc8e7cb7ec9ee8536597cfafb)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/execution/node/716/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/execution/node/828/log)

### History
* :broken_heart: [Build #138162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/) failed 83b6084f86b52defc9eab0ee543a8076e276febd
* :broken_heart: [Build #137804](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/) failed b8b3c9652af3fa0f0313ffe97ad40e523c917451
* :broken_heart: [Build #137794](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/) failed 90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/","number":"137794","commit":"90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/","number":"137804","commit":"b8b3c9652af3fa0f0313ffe97ad40e523c917451"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/","number":"138162","commit":"83b6084f86b52defc9eab0ee543a8076e276febd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/","number":"138173","commit":"733a971301d3a42cc8e7cb7ec9ee8536597cfafb"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/)
* Commit: a39dcf84ff9907ff0626ec9ffb27ecc667249587
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/a39dcf84ff9907ff0626ec9ffb27ecc667249587)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/1fa33ea751e18be2e27dd07d78142fb17bd77e76) | [after](https://github.com/elastic/kibana/commit/a39dcf84ff9907ff0626ec9ffb27ecc667249587) | diff |
| --- | --- | --- | --- |
| `apm` | 4.3MB | 4.3MB | +1.4KB |

### History
* :broken_heart: [Build #138173](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/) failed 733a971301d3a42cc8e7cb7ec9ee8536597cfafb
* :broken_heart: [Build #138162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/) failed 83b6084f86b52defc9eab0ee543a8076e276febd
* :broken_heart: [Build #137804](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/) failed b8b3c9652af3fa0f0313ffe97ad40e523c917451
* :broken_heart: [Build #137794](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/) failed 90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/","number":"137794","commit":"90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/","number":"137804","commit":"b8b3c9652af3fa0f0313ffe97ad40e523c917451"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/","number":"138162","commit":"83b6084f86b52defc9eab0ee543a8076e276febd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/","number":"138173","commit":"733a971301d3a42cc8e7cb7ec9ee8536597cfafb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/","number":"138369","commit":"a39dcf84ff9907ff0626ec9ffb27ecc667249587"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

walterra committed:
use 8.0.0 dataset instead of ml_8.0.0

walterra commented:
The dataset `ml_8.0.0` used for API integration tests (#104644 ) unfortunately didn't work on branches `7.x/7.14`. I switched the dataset to `8.0.0` and adjusted the assertions in those backports. dada021 is a cherry-pick of that change to bring it forward to `master` again as part of this PR.

pheyos commented on x-pack/test/functional/apps/apm/correlations/latency_correlations.ts:
```diff
@@ -0,0 +1,137 @@
+      it('loads the correlation results', async function () {
+        await retry.try(async () => {
+          // Assert that the data fully loaded to 100%
+          const apmCorrelationsFlyoutHeader = await testSubjects.getVisibleText(
+            'apmCorrelationsLatencyCorrelationsProgressTitle'
```
Nit: variable name doesn't really match the content. For the other element test variables I can see a pattern of naming the variable similar to the test subject, maybe that would fit here as well?

walterra replied:
Thanks for spotting that! Fixed in 2b2872c.

pheyos reviewed (approved):
Functional tests LGTM, just one small nit.

walterra committed:
fix variable names

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/)
* Commit: 2b2872c9c88c1f501d1fbfe4a2eade14f02ce431
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/2b2872c9c88c1f501d1fbfe4a2eade14f02ce431)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/1fa33ea751e18be2e27dd07d78142fb17bd77e76) | [after](https://github.com/elastic/kibana/commit/2b2872c9c88c1f501d1fbfe4a2eade14f02ce431) | diff |
| --- | --- | --- | --- |
| `apm` | 4.3MB | 4.3MB | +1.4KB |

### History
* :green_heart: [Build #138369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/) succeeded a39dcf84ff9907ff0626ec9ffb27ecc667249587
* :broken_heart: [Build #138173](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/) failed 733a971301d3a42cc8e7cb7ec9ee8536597cfafb
* :broken_heart: [Build #138162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/) failed 83b6084f86b52defc9eab0ee543a8076e276febd
* :broken_heart: [Build #137804](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/) failed b8b3c9652af3fa0f0313ffe97ad40e523c917451
* :broken_heart: [Build #137794](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/) failed 90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137794/","number":"137794","commit":"90b942c44ec4dc6ea7039a3d52c3f6b59b7fa9b5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/","number":"137804","commit":"b8b3c9652af3fa0f0313ffe97ad40e523c917451"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/","number":"138162","commit":"83b6084f86b52defc9eab0ee543a8076e276febd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/","number":"138173","commit":"733a971301d3a42cc8e7cb7ec9ee8536597cfafb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/","number":"138369","commit":"a39dcf84ff9907ff0626ec9ffb27ecc667249587"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/","number":"138694","commit":"2b2872c9c88c1f501d1fbfe4a2eade14f02ce431"}]}
PIPELINE-->

walterra committed:
Merge branch 'master' into ml-apm-correlations-functional-tests

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/)
* Commit: c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5fcdb9d137418b8a040d639e53b23b2a75108d0a) | [after](https://github.com/elastic/kibana/commit/c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe) | diff |
| --- | --- | --- | --- |
| `apm` | 4.3MB | 4.3MB | +1.4KB |

### History
* :green_heart: [Build #138694](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/) succeeded 2b2872c9c88c1f501d1fbfe4a2eade14f02ce431
* :green_heart: [Build #138369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/) succeeded a39dcf84ff9907ff0626ec9ffb27ecc667249587
* :broken_heart: [Build #138173](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/) failed 733a971301d3a42cc8e7cb7ec9ee8536597cfafb
* :broken_heart: [Build #138162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/) failed 83b6084f86b52defc9eab0ee543a8076e276febd
* :broken_heart: [Build #137804](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/) failed b8b3c9652af3fa0f0313ffe97ad40e523c917451

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137804/","number":"137804","commit":"b8b3c9652af3fa0f0313ffe97ad40e523c917451"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/","number":"138162","commit":"83b6084f86b52defc9eab0ee543a8076e276febd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/","number":"138173","commit":"733a971301d3a42cc8e7cb7ec9ee8536597cfafb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/","number":"138369","commit":"a39dcf84ff9907ff0626ec9ffb27ecc667249587"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/","number":"138694","commit":"2b2872c9c88c1f501d1fbfe4a2eade14f02ce431"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/","number":"138734","commit":"c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe"}]}
PIPELINE-->

qn895 commented on x-pack/plugins/apm/server/lib/search_strategies/correlations/async_search_service.ts:
```diff
@@ -28,7 +30,9 @@ const currentTimeAsString = () => new Date().toISOString();
-  params: SearchServiceParams
+  getApmIndices: () => Promise<ApmIndicesConfig>,
+  searchServiceParams: SearchServiceParams,
+  includeFrozen: boolean
```
Is `includeFrozen` being used here?

qn895 replied:
Just re-read the post description and my question is redundant here :) Sorry the noise

qn895 commented on x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx:
```diff
@@ -341,32 +348,50 @@ export function MlLatencyCorrelations({ onClose }: Props) {
+        ) : null}
+      </div>
+      {log.length > 0 && displayLog && (
+        <EuiAccordion id="accordion1" buttonContent="Log">
```
Does `Log` here needs internationalization?

walterra replied:
Fixed translation in 7592578.

qn895 commented:
Tested the debug mode and I really like it 🎉 . Definitely much more transparent. My only suggestion for a future follow up is to potentially log out the actual cause/error message as well instead of just `Failed to fetch ...`

On a super duper nit note, noticed the background color of the EuiPanel and the EuiCode are not the same, although I'm not sure if that's something worth adjusting.

walterra committed:
fix cross-cluster search error handling.

walterra committed:
fix passing on cssWarning.

walterra commented:
Added a fix to display a warning when we get errors in a cross-cluster environment in ecfd6e4.

![image](https://user-images.githubusercontent.com/230104/125965034-d905bba9-cb9e-431b-b21a-fa48838a0009.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/)
* Commit: 60bbed558fedb4068081e365ac9dd680d51a7353
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/execution/node/425/log)

### History
* :green_heart: [Build #138734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/) succeeded c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe
* :green_heart: [Build #138694](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/) succeeded 2b2872c9c88c1f501d1fbfe4a2eade14f02ce431
* :green_heart: [Build #138369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/) succeeded a39dcf84ff9907ff0626ec9ffb27ecc667249587
* :broken_heart: [Build #138173](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/) failed 733a971301d3a42cc8e7cb7ec9ee8536597cfafb
* :broken_heart: [Build #138162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/) failed 83b6084f86b52defc9eab0ee543a8076e276febd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138162/","number":"138162","commit":"83b6084f86b52defc9eab0ee543a8076e276febd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/","number":"138173","commit":"733a971301d3a42cc8e7cb7ec9ee8536597cfafb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/","number":"138369","commit":"a39dcf84ff9907ff0626ec9ffb27ecc667249587"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/","number":"138694","commit":"2b2872c9c88c1f501d1fbfe4a2eade14f02ce431"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/","number":"138734","commit":"c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/","number":"139080","commit":"60bbed558fedb4068081e365ac9dd680d51a7353"}]}
PIPELINE-->

walterra committed:
fix types.

walterra committed:
fix translation.

peteharverson commented on x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx:
```diff
@@ -313,11 +322,35 @@ export function MlLatencyCorrelations({ onClose }: Props) {
+            title={i18n.translate(
+              'xpack.apm.correlations.latencyCorrelations.ccsWarningCalloutTitle',
+              {
+                defaultMessage: 'Cross-cluster search compatilibity.',
```
Typo! Needs to be `compatibility`, and I'd remove the `.` as I think this is more in keeping with the writing guidelines.

walterra replied:
Fixes in 42a7cbd.

peteharverson reviewed (approved):
The log helper is nice! Just needs an edit on the CCS callout title, but otherwise LGTM

lcawl commented on x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx:
```diff
@@ -313,11 +322,35 @@ export function MlLatencyCorrelations({ onClose }: Props) {
+                'xpack.apm.correlations.latencyCorrelations.ccsWarningCalloutBody',
+                {
+                  defaultMessage:
+                    'Data for the correlation analysis could not be fully retrieved. This feature is only supported for versions 7.14 and above.',
```
Minor suggestion to use "later" instead of "above".
```suggestion
                    'Data for the correlation analysis could not be fully retrieved. This feature is supported only for 7.14 and later versions.',
```

walterra replied:
Fixed in 42a7cbd.

smith reviewed (approved):
I had a couple suggestions but this looks good.

We should plan to move these FTR E2E tests to our Cypress setup.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139096/)
* Commit: 75925780a9294d6eaf4c55940e29751dac0557fd
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/75925780a9294d6eaf4c55940e29751dac0557fd)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5fcdb9d137418b8a040d639e53b23b2a75108d0a) | [after](https://github.com/elastic/kibana/commit/75925780a9294d6eaf4c55940e29751dac0557fd) | diff |
| --- | --- | --- | --- |
| `apm` | 4.3MB | 4.3MB | +2.5KB |

### History
* :broken_heart: [Build #139080](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/) failed 60bbed558fedb4068081e365ac9dd680d51a7353
* :green_heart: [Build #138734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/) succeeded c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe
* :green_heart: [Build #138694](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/) succeeded 2b2872c9c88c1f501d1fbfe4a2eade14f02ce431
* :green_heart: [Build #138369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/) succeeded a39dcf84ff9907ff0626ec9ffb27ecc667249587
* :broken_heart: [Build #138173](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/) failed 733a971301d3a42cc8e7cb7ec9ee8536597cfafb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138173/","number":"138173","commit":"733a971301d3a42cc8e7cb7ec9ee8536597cfafb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/","number":"138369","commit":"a39dcf84ff9907ff0626ec9ffb27ecc667249587"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/","number":"138694","commit":"2b2872c9c88c1f501d1fbfe4a2eade14f02ce431"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/","number":"138734","commit":"c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/","number":"139080","commit":"60bbed558fedb4068081e365ac9dd680d51a7353"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139096/","number":"139096","commit":"75925780a9294d6eaf4c55940e29751dac0557fd"}]}
PIPELINE-->

walterra committed:
text fixes

walterra committed:
lint fix

walterra committed:
use observability:enableInspectEsQueries instead of debug=true

walterra committed:
Merge branch 'master' into ml-apm-correlations-functional-tests

walterra committed:
add retry to avoid flakyness.

walterra commented on x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx:
```diff
@@ -72,7 +75,11 @@ export function MlLatencyCorrelations({ onClose }: Props) {
   } = useUrlParams();
 
+  const location = useLocation();
+  const displayLog = location.search.includes('debug=true');
```
Great suggestion, fixed in 1a63809.

walterra commented on x-pack/plugins/apm/server/lib/search_strategies/correlations/search_strategy.test.ts:
```diff
@@ -94,10 +96,19 @@ const clientSearchMock = (
 
+const getApmIndicesMock = async () =>
+  ({
+    // eslint-disable-next-line
```
Fixed in 7d04ca3.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139287/)
* Commit: 5922e84c5a34bb45ad6e793ab37f941e65aff05e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-105200/5922e84c5a34bb45ad6e793ab37f941e65aff05e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/df1882eac025d2db7ca6cefbb1b19071d1b14888) | [after](https://github.com/elastic/kibana/commit/5922e84c5a34bb45ad6e793ab37f941e65aff05e) | diff |
| --- | --- | --- | --- |
| `apm` | 4.3MB | 4.3MB | +2.4KB |

### History
* :green_heart: [Build #139096](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139096/) succeeded 75925780a9294d6eaf4c55940e29751dac0557fd
* :broken_heart: [Build #139080](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/) failed 60bbed558fedb4068081e365ac9dd680d51a7353
* :green_heart: [Build #138734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/) succeeded c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe
* :green_heart: [Build #138694](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/) succeeded 2b2872c9c88c1f501d1fbfe4a2eade14f02ce431
* :green_heart: [Build #138369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/) succeeded a39dcf84ff9907ff0626ec9ffb27ecc667249587

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138369/","number":"138369","commit":"a39dcf84ff9907ff0626ec9ffb27ecc667249587"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138694/","number":"138694","commit":"2b2872c9c88c1f501d1fbfe4a2eade14f02ce431"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138734/","number":"138734","commit":"c439dbdcaf0172ccf2f129dea0cde88ec7ce7cbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139080/","number":"139080","commit":"60bbed558fedb4068081e365ac9dd680d51a7353"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139096/","number":"139096","commit":"75925780a9294d6eaf4c55940e29751dac0557fd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/139287/","number":"139287","commit":"5922e84c5a34bb45ad6e793ab37f941e65aff05e"}]}
PIPELINE-->
